File: app/src/main/java/com/example/krystianwsul/organizator/utils/time/TimePair.java
package com.example.krystianwsul.organizator.utils.time;

import android.os.Parcel;
import android.os.Parcelable;

import junit.framework.Assert;

public class TimePair implements Parcelable {
    public final Integer CustomTimeId;
    public final HourMinute HourMinute;

    public TimePair(Integer customTimeId, HourMinute hourMinute) {
        Assert.assertTrue((customTimeId == null) != (hourMinute == null));

        CustomTimeId = customTimeId;
        HourMinute = hourMinute;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel out, int flags) {
        if (CustomTimeId != null) {
            out.writeInt(1);
            out.writeInt(CustomTimeId);
        } else {
            out.writeInt(2);
            out.writeParcelable(HourMinute, 0);
        }
    }

    public static final Parcelable.Creator<TimePair> CREATOR = new Creator<TimePair>() {
        @Override
        public TimePair createFromParcel(Parcel source) {
            if (source.readInt() == 1) {
                return new TimePair(source.readInt(), null);
            } else {
                return new TimePair(null, (HourMinute) source.readParcelable(com.example.krystianwsul.organizator.utils.time.HourMinute.class.getClassLoader()));
            }
        }

        @Override
        public TimePair[] newArray(int size) {
            return new TimePair[size];
        }
    };
}

File: common/src/commonMain/kotlin/time/TimePair.kt
package com.krystianwsul.common.time

import com.krystianwsul.common.utils.CustomTimeKey
import com.krystianwsul.common.utils.Parcelable
import com.krystianwsul.common.utils.Parcelize
import com.krystianwsul.common.utils.Serializable

@Parcelize
data class TimePair(val customTimeKey: CustomTimeKey?, val hourMinute: HourMinute?) : Parcelable, Serializable {

    constructor(customTimeKey: CustomTimeKey) : this(customTimeKey, null)

    constructor(hourMinute: HourMinute) : this(null, hourMinute)

    init {
        check((customTimeKey == null) != (hourMinute == null))
    }
}
